flutter/lib/src/span_frame_metrics_collector.dart

```diff
@@ -0,0 +1,256 @@
+
+  bool get isTrackingPaused => _isTrackingPaused;
+  bool _isTrackingPaused = true;
+
+  bool get isTrackingRegistered => _isTrackingRegistered;
+  bool _isTrackingRegistered = false;
+
+  int displayRefreshRate = 60;
+
+  final _stopwatch = Stopwatch();
+
+  SpanFrameMetricsCollector(this.options,
+      {FrameCallbackHandler? frameCallbackHandler,
+      SentryNativeBinding? native,
+      @internal bool isTestMode = false})
+      : _frameCallbackHandler =
+            frameCallbackHandler ?? DefaultFrameCallbackHandler(),
+        _native = native ?? SentryFlutter.native,
+        _isTestMode = isTestMode;
+
+  @override
+  Future<void> onSpanStarted(ISentrySpan span) async {
+    if (span is NoOpSentrySpan || !options.enableFramesTracking) {
+      return;
+    }
+
+    final fetchedDisplayRefreshRate = await _native?.displayRefreshRate();
+    if (fetchedDisplayRefreshRate != null) {
+      options.logger(SentryLevel.debug,
+          'Retrieved display refresh rate at $fetchedDisplayRefreshRate');
+      displayRefreshRate = fetchedDisplayRefreshRate;
+    } else {
+      options.logger(SentryLevel.debug,
+          'Could not fetch display refresh rate, keeping at 60hz by default');
+    }
+
+    activeSpans.add(span);
+    startFrameTracking();
+  }
+
+  @override
+  Future<void> onSpanFinished(ISentrySpan span, DateTime endTimestamp) async {
+    if (span is NoOpSentrySpan || !activeSpans.contains(span)) return;
+
+    final frameMetrics =
+        calculateFrameMetrics(span, endTimestamp, displayRefreshRate);
+    _applyFrameMetricsToSpan(span, frameMetrics);
+
+    activeSpans.remove(span);
+    if (activeSpans.isEmpty) {
+      clear();
+    } else {
+      frames.removeWhere((frameTimestamp, _) =>
+          frameTimestamp.isBefore(activeSpans.first.startTimestamp));
+    }
+  }
+
+  /// Calls [WidgetsBinding.instance.addPersistentFrameCallback] which cannot be unregistered
+  /// and exists for the duration of the application's lifetime.
+  ///
+  /// Stopping the frame tracking means setting [isTrackingPaused] is `true`
+  /// to prevent actions being done when the frame callback is triggered.
+  void startFrameTracking() {
+    _isTrackingPaused = false;
+
+    if (!_isTrackingRegistered) {
+      _frameCallbackHandler.addPersistentFrameCallback(measureFrameDuration);
+      _isTrackingRegistered = true;
+    }
+  }
+
+  /// Records the duration of a single frame and stores it in [frames].
+  ///
+  /// This method is called for each frame when frame tracking is active.
+  Future<void> measureFrameDuration(Duration duration) async {
+    // Using the stopwatch to measure the frame duration is flaky in ci
+    if (_isTestMode) {
+      // ignore: invalid_use_of_internal_member
+      frames[options.clock().add(duration)] = duration.inMilliseconds;
+      return;
+    }
+
+    if (_isTrackingPaused) return;
+
+    if (!_stopwatch.isRunning) {
+      _stopwatch.start();
+    }
+
+    await _frameCallbackHandler.endOfFrame;
```

Comment:
@buenaflor 
Sorry for commenting after merge pr🙏
If you use "await" in a FrameCallback function, is there any possibility of delaying all actions?

Reply:
@buenaflor 
In our product, we experienced an issue with delayed screen transition behaviour, which improved when 'enableFramesTracking' was set to false. This is probably due to blocked processing in the FrameCallback.

At 60 fps, this callback is called up to 60 times per second.
You should not use await in FrameCallback!

Reply:
I see, do you have a way to reproduce this in a sample app? 

I'll have a look how we can optimize this

Reply:
> At 60 fps, this callback is called up to 60 times per second. You should not use await in FrameCallback!

this is not the reason. using await here is fine.

See this [issue](https://github.com/getsentry/sentry-dart/issues/2298) for info why this is happening